.pre-commit-config.yaml

```diff
@@ -79,3 +80,16 @@ repos:
     rev: "2.4.0"
     hooks:
       - id: nb-clean
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.991
+    hooks:
+      - id: mypy
+        files: ^(src|tests|setup.py)
+        args: []
+        additional_dependencies:
+          - types-setuptools
+          - numpy
+          - python_tsp
+          - networkx
```

Comment:
here we could think about switching to rustworkx (rust port of networkx developed by one of the Qiskit maintainers).
Would allow us to remove the dependency here and (most likely) improve the performance of all graph related stuff.
Nothing for this PR though. Just putting the idea out there.